File: app/assets/javascripts/app/models/group_buy.js
UniteToBuyIt.GroupBuy = DS.Model.extend({
  createdAt:        DS.attr('string'),
  updatedAt:        DS.attr('string'),

  userId: DS.belongsTo('UniteToBuyIt.User', {
    inverse: 'groupBuyIds'
  }),
  name:             DS.attr('string'),
  email:            DS.attr('string'),
  // organizer:        DS.belongsTo('UniteToBuyIt.User'),

  manufacturer:     DS.attr('string'),

  endsAt:           DS.attr('string'),
  deliverableAt:    DS.attr('string'),
  cancelledAt:      DS.attr('string'),

  introduction:     DS.attr('string'),
  statusUpdates:    DS.attr('string'),

  orders: DS.hasMany('UniteToBuyIt.Order'),
  orderableSets: DS.hasMany('UniteToBuyIt.OrderableSet'),
  statusUpdates: DS.hasMany('UniteToBuyIt.StatusUpdate'),
  introduction: DS.belongsTo('UniteToBuyIt.Introduction'),


  orderableSets:    DS.attr('string')
  // orders:           DS.attr('string')
});

File: app/models/group_buy.rb
class GroupBuy
  include Mongoid::Document
  include Mongoid::Timestamps

  ################################
  # Mongoid relation definitions #
  ################################

  belongs_to  :user

  embeds_one  :introduction, autobuild: true
  embeds_many :status_updates

  has_many :orders

  has_many :orderable_sets


  #############################
  # Mongoid field definitions #
  #############################

  # Group Buy information fields

  field :name                   , type: String
  field :email                  , type: String
  field :manufacturer           , type: String

  # Descriptions and updates

# field :introduction           , type: Object (embedded document)
# field :status_updates         , type: Object (embedded document)

  # Related community links

  field :deskthority_thread_url , type: String
  field :geekhack_thread_url    , type: String
  field :reddit_thread_url      , type: String

  # Timekeeping

  field :ends_at                , type: DateTime
  field :deliverable_at         , type: DateTime
  field :cancelled_at           , type: DateTime





  field :payment_collectors     , type: Array

  field :orderable_sets         , type: Array

  field :orders                 , type: Array


end
